### ISSUE TITLE
Please provide a working SMI example

### ISSUE THREAD

**Issue by rkr-kununu** [containous/maesh#412]
#### Proposal

I'm really excited about the SMI support that Maesh has.  However, I'm unable to get a very simple example to work.  A fully functional example would be very helpful.

#### Background

I've installed Maesh v1.1-rc0 with helm (v3.0.2) by running:
`helm install maesh --namespace=maesh maesh/maesh --set smi.enable=true`

Then I followed the instructions outlined here: https://docs.mae.sh/v1.1/examples/

Afterward, I tried to create the proper SMI resources:
```
---
apiVersion: specs.smi-spec.io/v1alpha1
kind: HTTPRouteGroup
metadata:
  name: http-everything
  namespace: whoami
matches:
- name: everything
  pathRegex: ".*"
  methods: ["*"]
---
kind: TrafficTarget
apiVersion: access.smi-spec.io/v1alpha1
metadata:
  name: whatever
  namespace: whoami
destination:
  kind: ServiceAccount
  name: whoami-server
  namespace: whoami
  port: 80
specs:
- kind: HTTPRouteGroup
  name: http-everything
  matches:
  - everything
sources:
- kind: ServiceAccount
  name: whoami-client
  namespace: whoami
```

Unfortunately, when I run:
`kubectl -n whoami exec whoami-client -- curl -s whoami.whoami.maesh`

I get a HTTP-404.

#### Workarounds

None I know of.

**Comment by dtomcej** [comment]
Hello @rkr-kununu,

Can you confirm that the maesh controller pod is ready?

Can you provide either the current configuration from the controller?

It can be accessed via port 9000, and `/api/configuration/current`.

https://docs.mae.sh/v1.1/api/

This will allow us to confirm what configuration is being parsed in your environment.

Thanks!

**Comment by dtomcej** [comment]
(Of note, we still should have a working SMI example in the docs, regardless of if we are able to get @rkr-kununu's issue resolved)

**Comment by rkr-kununu** [comment]
Thank you for the help.  I can confirm that everything that's launch is running:

```
$ kubectl -n maesh get all
NAME                                    READY   STATUS    RESTARTS   AGE
pod/jaeger-787f575fbd-kqkrk             1/1     Running   0          7m
pod/prometheus-core-996d966fb-v5b6p     2/2     Running   0          7m
pod/maesh-controller-6496f567d9-n74qb   1/1     Running   0          7m
pod/grafana-core-7cb56c69cb-7pqdf       1/1     Running   0          7m
pod/maesh-mesh-8twgb                    1/1     Running   0          7m

NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                               AGE
service/grafana                   NodePort    10.43.195.255   <none>        3000:31589/TCP                        7m
service/prometheus                NodePort    10.43.206.72    <none>        9090:31679/TCP                        7m
service/zipkin                    ClusterIP   None            <none>        9411/TCP                              7m
service/jaeger-agent              ClusterIP   None            <none>        5775/UDP,6831/UDP,6832/UDP,5778/TCP   7m
service/jaeger-collector          ClusterIP   10.43.82.24     <none>        14267/TCP,14268/TCP,9411/TCP          7m
service/jaeger-query              ClusterIP   10.43.178.43    <none>        16686/TCP                             7m
service/maesh-mesh-api            ClusterIP   10.43.155.253   <none>        8080/TCP                              7m
service/maesh-whoami-whoami       ClusterIP   10.43.172.220   <none>        80/TCP                                5m
service/maesh-whoami-tcp-whoami   ClusterIP   10.43.223.171   <none>        8080/TCP                              5m

NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/maesh-mesh   1         1         1       1            1           <none>          7m

NAME                               READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/jaeger             1/1     1            1           7m
deployment.apps/prometheus-core    1/1     1            1           7m
deployment.apps/maesh-controller   1/1     1            1           7m
deployment.apps/grafana-core       1/1     1            1           7m

NAME                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/jaeger-787f575fbd             1         1         1       7m
replicaset.apps/prometheus-core-996d966fb     1         1         1       7m
replicaset.apps/maesh-controller-6496f567d9   1         1         1       7m
replicaset.apps/grafana-core-7cb56c69cb       1         1         1       7m
```

Furthermore, here's the configuration of the controller

```
$ wget -qO- http://localhost:9000/api/configuration/current
{"http":{"routers":{"readiness":{"entryPoints":["readiness"],"service":"readiness","rule":"Path(`/ping`)"}},"middlewares":{"smi-block-all-middleware":{"ipWhiteList":{"sourceRange":["255.255.255.255"]}}},"services":{"readiness":{"loadBalancer":{"servers":[{"url":"http://127.0.0.1:8080"}],"passHostHeader":true}}}},"tcp":{}}
```

What other information can I provide to you?

**Comment by rkr-kununu** [comment]
Here is a snippet of the logs from the maesh-controller.  Maybe you'll find them useful:

```
time="2020-02-10T13:45:21Z" level=debug msg="Updated service: maesh/maesh-whoami-whoami"
time="2020-02-10T13:45:21Z" level=debug msg="MeshControllerHandler ObjectUpdated with type: *corev1.Service: whoami/whoami"
time="2020-02-10T13:45:21Z" level=debug msg="MeshControllerHandler ObjectUpdated with type: *corev1.Endpoints: whoami/whoami"
time="2020-02-10T13:45:21Z" level=debug msg="MeshControllerHandler ObjectUpdated with type: *corev1.Pod: maesh/maesh-mesh-8twgb"
time="2020-02-10T13:45:21Z" level=debug msg="Updated service: maesh/maesh-whoami-tcp-whoami"
time="2020-02-10T13:45:21Z" level=debug msg="MeshControllerHandler ObjectUpdated with type: *corev1.Service: whoami/whoami-tcp"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami-tcp: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="MeshControllerHandler ObjectUpdated with type: *corev1.Endpoints: whoami/whoami-tcp"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami-tcp: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami-tcp: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Deploying to pod maesh-mesh-8twgb with IP 10.42.0.5"
time="2020-02-10T13:45:21Z" level=debug msg="Successfully deployed configuration to pod (maesh-mesh-8twgb:10.42.0.5)"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found applicable traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:21Z" level=debug msg="Found grouped traffictargets for service whoami/whoami-tcp: map[]\n"
time="2020-02-10T13:45:21Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:21Z" level=debug msg="Found trafficsplits for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:22Z" level=debug msg="Found traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="Found applicable traffictargets for service whoami/whoami: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="Found grouped traffictargets for service whoami/whoami: map[]\n"
time="2020-02-10T13:45:22Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:22Z" level=debug msg="Found trafficsplits for service whoami/whoami: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="No TrafficTargets with destination in namespace: whoami"
time="2020-02-10T13:45:22Z" level=debug msg="Found traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="Found applicable traffictargets for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:22Z" level=debug msg="Found grouped traffictargets for service whoami/whoami-tcp: map[]\n"
time="2020-02-10T13:45:22Z" level=debug msg="No TrafficSplits in namespace: whoami"
time="2020-02-10T13:45:22Z" level=debug msg="Found trafficsplits for service whoami/whoami-tcp: []\n"
time="2020-02-10T13:45:32Z" level=debug msg="Deploying configuration to unready nodes"
time="2020-02-10T13:45:42Z" level=debug msg="Deploying configuration to unready nodes"
time="2020-02-10T13:45:52Z" level=debug msg="Deploying configuration to unready nodes"
time="2020-02-10T13:46:02Z" level=debug msg="Deploying configuration to unready nodes"
time="2020-02-10T13:46:12Z" level=debug msg="Deploying configuration to unready nodes"
```

**Comment by dtomcej** [comment]
@rkr-kununu I have confirmed your issue with the string port configuration.

I am going to open a new issue and link it here.

**Comment by traefiker** [comment]
Closed by #432.
